src/main/scala/uk/gov/hmrc/ct/ct600a/v3/retriever/CT600ABoxRetriever.scala

```diff
@@ -16,14 +16,13 @@
 
 package uk.gov.hmrc.ct.ct600a.v3.retriever
 
-import uk.gov.hmrc.ct.accounts.retriever.AccountsBoxRetriever
+import uk.gov.hmrc.ct.box.retriever.BoxRetriever
 import uk.gov.hmrc.ct.computations.retriever.ComputationsBoxRetriever
 import uk.gov.hmrc.ct.ct600.v3.retriever.{AboutThisReturnBoxRetriever, CT600BoxRetriever}
 import uk.gov.hmrc.ct.ct600a.v3._
 
-trait CT600ABoxRetriever extends ComputationsBoxRetriever {
-
-  self: CT600BoxRetriever with AccountsBoxRetriever with AboutThisReturnBoxRetriever =>
+abstract class CT600ABoxRetriever(val computationsBoxRetriever: ComputationsBoxRetriever,
+                                  val ct600BoxRetriever: CT600BoxRetriever) extends BoxRetriever {
```

Comment:
Yeah, I prefer this style of injecting the boxes and using their values, as discussed above. What do you think?